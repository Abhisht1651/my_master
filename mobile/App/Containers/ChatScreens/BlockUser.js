// @flow
import React from 'react'
import { View, Text, Modal, TouchableOpacity, Image } from 'react-native'
import Images from '../../Themes/Images'
import styles from './ScreenStyles/BlockUserStyles'

type Props = {
  isModalVisible: boolean,
  closeAction: () => void,
  blockAction: () => void
}
const BlockProfilePopUp = (props: Props) => {
  const { isModalVisible, closeAction, blockAction } = props
  return (
    <Modal visible={isModalVisible} transparent>
      <View style={styles.modalStyle}>
        <View style={styles.card}>
          <TouchableOpacity
            onPress={closeAction}
            style={styles.closeIconContainer}
            activeOpacity={1}>
            <Image source={Images.closeIcon} style={styles.imageStyle} />
          </TouchableOpacity>
          <View style={styles.textContainer}>
            <Text style={styles.text1}>Block Profile</Text>
            <Text style={styles.text2}>Are you sure you want to block this profile?</Text>
            <Text style={styles.infoText}>
              They won’t be able to see your posts, activities or stories. Newzera won’t let them
              know you blocked them.
            </Text>
          </View>
          <TouchableOpacity
            onPress={blockAction}
            style={styles.blockButtonContainer}
            activeOpacity={1}>
            <Text style={styles.blockButtonText}>Block</Text>
          </TouchableOpacity>
        </View>
      </View>
    </Modal>
  )
}

export default BlockProfilePopUp
