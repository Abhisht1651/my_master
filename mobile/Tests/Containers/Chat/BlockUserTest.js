import React from 'react'
import { shallow } from 'enzyme'
import { TouchableOpacity, Text, Image } from 'react-native'
import BlockUser from '../../../App/Containers/ChatScreens/BlockUser'

describe('BlockUser', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<BlockUser />) // no compile-time error
  })
  it('BlockUser component renders correctly', () => {
    const wrapper = shallow(<BlockUser />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render 2 TouchableOpacity', () => {
    expect(wrapper.find(TouchableOpacity)).toHaveLength(2)
  })
  it('should render 4 Text', () => {
    expect(wrapper.find(Text)).toHaveLength(4)
  })
  it('should render 1 Image', () => {
    expect(wrapper.find(Image)).toHaveLength(1)
  })
})
