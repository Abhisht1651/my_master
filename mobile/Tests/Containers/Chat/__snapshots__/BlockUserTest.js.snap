// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockUser BlockUser component renders correctly 1`] = `
<Component
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2222234D",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 15,
          "height": 329,
          "maxWidth": "86.66%",
          "width": "86.66%",
        }
      }
    >
      <ForwardRef
        activeOpacity={1}
        style={
          Object {
            "alignItems": "flex-end",
            "height": 19,
            "justifyContent": "flex-end",
            "marginTop": 23.25,
            "width": "84%",
          }
        }
      >
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "height": 19,
              "tintColor": "#1E1E60",
              "width": 19,
            }
          }
        />
      </ForwardRef>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": 3.97,
            "width": "82.71%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 30,
              "letterSpacing": -0.14,
            }
          }
        >
          Block Profile
        </Text>
        <Text
          style={
            Object {
              "color": "#262676",
              "fontFamily": "BananaGrotesk-Light",
              "fontSize": 16,
              "height": 45,
              "letterSpacing": -0.11,
              "lineHeight": 22.5,
              "marginTop": 6,
              "textAlign": "center",
              "width": 170,
            }
          }
        >
          Are you sure you want to block this profile?
        </Text>
        <Text
          style={
            Object {
              "color": "#8F8FAF",
              "fontFamily": "BananaGrotesk-Light",
              "fontSize": 14,
              "height": 70,
              "lineHeight": 23,
              "marginTop": 23,
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          They won’t be able to see your posts, activities or stories. Newzera won’t let them know you blocked them.
        </Text>
      </View>
      <ForwardRef
        activeOpacity={1}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#7B46E4",
            "borderRadius": 39,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 38,
            "width": "84%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 16,
              "letterSpacing": -0.09,
            }
          }
        >
          Block
        </Text>
      </ForwardRef>
    </View>
  </View>
</Component>
`;
